### PR TITLE
Add a reactive version of the Android.Support.V4.DialogFragment class

### PR DESCRIPTION
**Pull request by flagbug**

### CODE REVIEW

File: ReactiveUI.AndroidSupport/ReactiveDialogFragment.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+namespace ReactiveUI.AndroidSupport
+{
+    /// <summary>
+    /// This is a DialogFragment that is both a DialogFragment and has ReactiveObject powers 
+    /// (i.e. you can call RaiseAndSetIfChanged)
+    /// </summary>
+    public class ReactiveDialogFragment<TViewModel> : ReactiveDialogFragment, IViewFor<TViewModel>, ICanActivate
+        where TViewModel : class
+    {
+        protected ReactiveDialogFragment() { }
```
Comment (kentcb) [inline ReactiveUI.AndroidSupport/ReactiveDialogFragment.cs]: Is there a reason the ctor is protected but the class is not abstract? Same question for the non-generic counterpart.

Comment (flagbug) [inline ReactiveUI.AndroidSupport/ReactiveDialogFragment.cs]: Yeah, you're right, I'll fix that

Comment (flagbug) [inline ReactiveUI.AndroidSupport/ReactiveDialogFragment.cs]: @kentcb I just looked at the other fragments (i.e `ReactiveUI.AndroidSupport.DialogFragment`) and they're also not marked as `abstract`

Comment (kentcb) [inline ReactiveUI.AndroidSupport/ReactiveDialogFragment.cs]: Fair enough - broken windows theory and all that ;)

**Pull request merged**